Xamarin.Forms.Platform.Android/Renderers/ShellSectionRenderer.cs

```diff
@@ -48,7 +48,11 @@ void ViewPager.IOnPageChangeListener.OnPageSelected(int position)
 			{
 				ShellSection.SetValueFromRenderer(ShellSection.CurrentItemProperty, shellContent);
 
-				_toolbarTracker.Page = ((IShellContentController)shellContent).Page;
+				var page = ((IShellContentController)shellContent).Page;
+				if(page == null)
+					throw new ArgumentNullException(nameof(page), "Shell Content Page is Null");
```

Comment:
is this a valid check? This bug was annoying to track backwards to here so I put this here. ToolbarTracker can't really function with page set to null

Reply:
Should it be earlier? If it's going to fail anyway, should ProposeNavigation even be called?

Reply:
I moved it above the *SetValueFromRenderer* call but I think it's still fine there. That page is only used in that branch of the if statement so it seems excessive to have it always do that check in a case where the result is going to be false